src/main/java/dev/morling/onebrc/CreateMeasurementsFast.java

```diff
@@ -0,0 +1,533 @@
+                new WeatherStation("Tashkent", 14.8),
+                new WeatherStation("Tauranga", 14.8),
+                new WeatherStation("Tbilisi", 12.9),
+                new WeatherStation("Tegucigalpa", 21.7),
+                new WeatherStation("Tehran", 17.0),
+                new WeatherStation("Tel Aviv", 20.0),
+                new WeatherStation("Thessaloniki", 16.0),
+                new WeatherStation("Thiès", 24.0),
+                new WeatherStation("Tijuana", 17.8),
+                new WeatherStation("Timbuktu", 28.0),
+                new WeatherStation("Tirana", 15.2),
+                new WeatherStation("Toamasina", 23.4),
+                new WeatherStation("Tokyo", 15.4),
+                new WeatherStation("Toliara", 24.1),
+                new WeatherStation("Toluca", 12.4),
+                new WeatherStation("Toronto", 9.4),
+                new WeatherStation("Tripoli", 20.0),
+                new WeatherStation("Tromsø", 2.9),
+                new WeatherStation("Tucson", 20.9),
+                new WeatherStation("Tunis", 18.4),
+                new WeatherStation("Ulaanbaatar", -0.4),
+                new WeatherStation("Upington", 20.4),
+                new WeatherStation("Ürümqi", 7.4),
+                new WeatherStation("Vaduz", 10.1),
+                new WeatherStation("Valencia", 18.3),
+                new WeatherStation("Valletta", 18.8),
+                new WeatherStation("Vancouver", 10.4),
+                new WeatherStation("Veracruz", 25.4),
+                new WeatherStation("Vienna", 10.4),
+                new WeatherStation("Vientiane", 25.9),
+                new WeatherStation("Villahermosa", 27.1),
+                new WeatherStation("Vilnius", 6.0),
+                new WeatherStation("Virginia Beach", 15.8),
+                new WeatherStation("Vladivostok", 4.9),
+                new WeatherStation("Warsaw", 8.5),
+                new WeatherStation("Washington, D.C.", 14.6),
+                new WeatherStation("Wau", 27.8),
+                new WeatherStation("Wellington", 12.9),
+                new WeatherStation("Whitehorse", -0.1),
+                new WeatherStation("Wichita", 13.9),
+                new WeatherStation("Willemstad", 28.0),
+                new WeatherStation("Winnipeg", 3.0),
+                new WeatherStation("Wrocław", 9.6),
+                new WeatherStation("Xi'an", 14.1),
+                new WeatherStation("Yakutsk", -8.8),
+                new WeatherStation("Yangon", 27.5),
+                new WeatherStation("Yaoundé", 23.8),
+                new WeatherStation("Yellowknife", -4.3),
+                new WeatherStation("Yerevan", 12.4),
+                new WeatherStation("Yinchuan", 9.0),
+                new WeatherStation("Zagreb", 10.7),
+                new WeatherStation("Zanzibar City", 26.0),
+                new WeatherStation("Zürich", 9.3));
+
+        int chunkSize = 10_000_000;
+        int numberOfFutures = size / chunkSize;
```

Comment:
This rounds down, so trying to produce fewer than 10_000_000 records gives an empty output (because 0 futures are spawned). 

Additionally, when I run this I consistently get fewer lines than requested (e.g. 9_999_999 when I ask for 10_000_000, 99_999_990 when I ask for 100_000_000). Not sure where it comes from, but could you fix that?

Reply:
yeah. just now fixed the issue with correct number of lines when generating 1B lines.
btw, why should you generate less than that? challenge is 1B rows ;)

Reply:
anyways, here the fix for less or equal to 10M entries of the chunk size @datdenkikniet

Reply:
Nice :smile: